File: week08/4779.js
// 2024.05.23 풀이
// 칸토어 집합
// https://www.acmicpc.net/problem/4779

const fs = require("fs");
const path = "./input.txt";

const readline = require("readline");
const rl = readline.createInterface({
	// input: process.stdin,
	input: fs.createReadStream(path),
	output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
	input.push(line);
}).on("close", function () {
	solution(input);
	process.exit();
});

function solution(input) {
	input.forEach((n) => {
		let result = "-";
		for (let i = 0; i < +n; i++)
			result = result + " ".repeat(result.length) + result;
		console.log(result);
	});
}

File: [8주차]/4779.py
# 8주차 복습하기
from sys import stdin
input = stdin.readline

def cantor(n):
    if n == 1:
        return "-"

    unit = cantor(n // 3)
    res = unit + " " * (n // 3) + unit

    return res

while True:
    try:
        N = int(input())
        print(cantor(3**N))
    except:
        break